app/upgrades/v5/upgrades_test.go

```diff
@@ -298,3 +300,134 @@ func (suite *UpgradeTestSuite) TestUpdateConsensusParams() {
 		})
 	}
 }
+
+func (suite *UpgradeTestSuite) TestUpdateIBCDenomTraces() {
+	testCases := []struct {
+		name           string
+		malleate       func()
+		expDenomTraces ibctransfertypes.Traces
+	}{
+		{
+			"no traces",
+			func() {},
+			ibctransfertypes.Traces{},
+		},
+		{
+			"native IBC tokens",
+			func() {
+				traces := ibctransfertypes.Traces{
+					{
+						BaseDenom: "aevmos",
+						Path:      "",
+					},
+					{
+						BaseDenom: "uosmo",
+						Path:      "transfer/channel-0",
+					},
+					{
+						BaseDenom: "uatom",
+						Path:      "transfer/channel-0/transfer/channel-0",
+					},
+					{
+						BaseDenom: "uatom",
+						Path:      "transfer/channel-3",
+					},
+					{
+						BaseDenom: "gravity0x6B175474E89094C44Da98b954EedeAC495271d0F",
+						Path:      "transfer/channel-8",
+					},
+				}
+				for _, dt := range traces {
```

Comment:
instead of repeating this iteration in `malleate` you could add it within `for _, tc := range testCases {`